src/compiler/utilities.ts

```diff
@@ -6903,4 +6903,77 @@ namespace ts {
             return bindParentToChildIgnoringJSDoc(child, parent) || bindJSDoc(child);
         }
     }
+
+    export function getPossibleSymbolReferencePositions(sourceFile: SourceFile, symbolName: string, container: Node = sourceFile) {

```

Comment:
This won't work on something with Unicode escapes. It's not a blocker, but it'd be very strange.

```ts
new Promise(resolve => {
    \u0072\u0065\u0073\u006f\u006c\u0076\u0065(100);
})
```

Reply:
True, and that's documented in the find-all-references code where this function originally came from. Its unfortunate, but this heuristic is intended to improve inference in a very specific case. It might be feasible to rewrite the function to be a bit smarter (though possibly slower), by scanning the text instead of using `.indexOf`. That would likely help with find-all-references as well.

Reply:
(Though I'm not sure a reasonable change based on scanning could be made prior to 4.1-beta)

Reply:
Actually, scratch that. Its feasible to do. The implementation that supports unicode escapes depends on using a regular expression, so is costlier. I have a version I can push that actually tracks on the source file whether it contains identifiers with unicode escapes, and if it doesn't then it can use the existing `indexOf` logic, which should be faster:

![image](https://user-images.githubusercontent.com/3902892/92836133-ec171f80-f390-11ea-8ada-935d4f99aee1.png)

I might wait on that until after this PR, however, as I'd need to also add fourslash tests for find-all-references which could be time consuming and could possibly wait until after 4.1-beta.